cylc/flow/suite_files.py

```diff
@@ -669,3 +667,37 @@ def _load_local_item(item, path):
             return file_.read()
     except IOError:
         return None
+
+
+def check_nested_run_dirs(reg):
+    """Disallow nested run dirs e.g. trying to register foo/bar where foo is
+    already a valid suite directory.
+
+    Args:
+        reg (str): suite name
+    """
+    parent_dir = os.path.dirname(os.path.normpath(reg))
+    while parent_dir != '':
+        if _is_valid_run_dir(parent_dir):
+            raise SuiteServiceFileError(
+                'Nested run directories not allowed - cannot register suite '
+                f'name "{reg}" as {parent_dir} is already a valid '
+                'run directory.')
+        parent_dir = os.path.dirname(parent_dir)
+
+
+def _is_valid_run_dir(path):
+    """Return True if path is a valid suite run directory, else False.
+
+    Args:
+        path (str): if this is a relative path, it is taken to be relative to
+            the Cylc run directory.
+    """
+    if not os.path.isabs(path):
+        cylc_run_dir = os.path.expandvars(get_platform()['run directory'])
+        path = os.path.join(cylc_run_dir, path)
+    if (os.path.isfile(os.path.join(path, SuiteFiles.FLOW_FILE)) or
+            os.path.isdir(os.path.join(path, SuiteFiles.Service.DIRNAME)) or
+            os.path.isfile(os.path.join(path, SuiteFiles.SUITE_RC))):
+        return True
```

Comment:
Instead of checking for the existence of the `.service` dir, should we check that `.sevice/source` is an existing symlink?

Reply:
Existing or valid?

Are you suggesting we disallow flows inside existing valid run directories, but we allow them inside existing invalid ones?

That seems untidy at best, to me (is there is use case?).

Reply:
I was wondering whether testing for `.service/source` was a more appropriate gauge of whether the directory was valid, but I guess just `.service` is good enough

Reply:
Actually checking for `flow.cylc` or `suite.rc` or `.service/` as above might be overkill.  Detecting `.service/` is probably sufficient isn't it?  I would say the presence of a service directory is really what defines a Cylc run directory, not the presence of a flow config file, which can appear in other places too.

Reply:
I see... Though checking for `flow.cylc` will catch if you tried to register a suite in a sub dir of an unregistered suite. But come to think of it, I suppose that isn't desirable, actually?

Reply:
I've pushed a new commit removes the check for `flow.cylc` / `suite.rc`